src/utils/chat.ts

```diff
@@ -60,26 +32,29 @@ export const getPreviewChatMessage = ({
   counterPartyUser: user,
 });
 
-export interface GetPreviewChatMessageToDisplayProps {
-  message: PreviewChatMessage;
-  myUserId?: string;
-}
-
-export const getPreviewChatMessageToDisplay = ({
-  message,
-  myUserId,
-}: GetPreviewChatMessageToDisplayProps): PreviewChatMessageToDisplay => ({
-  ...message,
-  isMine: myUserId === message.from,
-});
-
 export const buildMessage = <T extends ChatMessage>(
-  message: Pick<T, Exclude<keyof T, "ts_utc">>
+  from: WithId<ChatUser>,
+  message: Pick<T, Exclude<keyof T, "ts_utc" | "from">>
 ) => ({
   ...message,
+  from: pickChatUserFromUser(from),
```

Comment:
I guess you already pass chat user?

Reply:
That's for safety. 
The type of `from` is indeed `WithId<ChatUser>` but you can pass whenever "descendant" of that type. `WithId<User>` is usually passed but I didn't want to restrict to that as we only need `ChatUser` fields from `User`.
 But the object built here is sent to Firebase so we need to ensure that only needed fields are present in the object.